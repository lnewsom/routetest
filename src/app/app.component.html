<!--The content below is only a placeholder and can be replaced.-->
<div style="text-align:center">
  <mat-toolbar>This header is on every page</mat-toolbar>
  <router-outlet></router-outlet>
