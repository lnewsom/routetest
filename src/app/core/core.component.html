<app-header [coreId]="this.coreId"></app-header>
<router-outlet></router-outlet>
