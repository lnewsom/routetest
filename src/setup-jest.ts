import 'jest-preset-angular';
import './setup-jest';
