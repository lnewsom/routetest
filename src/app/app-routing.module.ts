import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { CoreComponent } from './core/core.component';
import { FeatureComponent } from './feature/feature.component';

const routes: Routes = [
  {path: 'core/:coreId', component: CoreComponent,
  children: [
    { path: 'feature' , component: FeatureComponent}
  ]}
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
