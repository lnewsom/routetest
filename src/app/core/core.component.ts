import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-core',
  templateUrl: './core.component.html',
  styleUrls: ['./core.component.scss']
})
export class CoreComponent implements OnInit {

  constructor(private route: ActivatedRoute) { }

  ngOnInit() {
    this.route.parent.params.subscribe((params) => {
      console.log('Core Params: ' , params);
    });
  }

}
