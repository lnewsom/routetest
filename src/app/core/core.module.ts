import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CoreComponent } from './core.component';
import { HeaderComponent } from './header/header.component';
import { AppRoutingModule } from '../app-routing.module';


@NgModule({
  declarations: [
    CoreComponent,
    HeaderComponent
  ],
  imports: [
    CommonModule,
    AppRoutingModule
    ], exports: [
    HeaderComponent
  ]
})
export class CoreModule { }
