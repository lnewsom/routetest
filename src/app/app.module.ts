import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { MatToolbarModule } from '@angular/material/toolbar';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { CoreModule } from './core/core.module';
import { FeatureModule } from './feature/feature.module';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    CoreModule,
    MatToolbarModule,
    FeatureModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
