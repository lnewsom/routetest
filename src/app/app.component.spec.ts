import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';

jest.mock('@angular/core');

describe('AppComponent', () => {
  describe('snapshot', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        declarations: [
          AppComponent
        ],
        schemas: [NO_ERRORS_SCHEMA]
      }).compileComponents();
    }));

  test('should match snapshot', () => {
    const fixture: ComponentFixture<AppComponent> = TestBed.createComponent(AppComponent);

    expect(fixture).toMatchSnapshot();

  });
  });
});
