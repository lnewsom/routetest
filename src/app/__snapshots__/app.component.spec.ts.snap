// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot should match snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <div
    style="text-align:center"
  >
    <mat-toolbar>
      This header is on every page
    </mat-toolbar>
    <router-outlet />
  </div>
</app-root>
`;
